File: Java/Searching Algorithms/BinarySearch.java
public class BinarySearch {

    public static int search(int[] arr, int key) {
        int l = 0, r = arr.length - 1;
        while (l <= r) {
            int m = l + (r - l) / 2;

            if (arr[m] == key)
                return m;

            if (arr[m] < key)
                l = m + 1;
            else
                r = m - 1;
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {2, 3, 4, 10, 40};
        int searchKey = 10;

        int result = search(arr, searchKey);
        System.out.print(result == -1 ? "Element is not present in array" : "Element is present at index " + result);
    }
}

File: javascript/binarySearch.js
function BinarySearch(arr, ele) {
     let lo = 0, hi = arr.length - 1;
     while(lo <= hi) {
         let mid = lo + (hi - lo) / 2;
         if(arr[mid] == ele) {
             return console.log(mid);
         }
         else if(arr[mid] > ele) {
            hi = mid - 1;
         }
         else lo = mid + 1;
     }
     return console.log('Element not found');
  }
  
  var arr = [234, 43, 55, 63, 5, 6, 235, 547]
  
  BinarySearch(arr, 5)
  